src/Components/GraduatesList/GraduatesList.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from "react";
+import { Checkbox, Input, Table } from "semantic-ui-react";
+import FilterButton from "./FilterButton";
+import Graduated from "./Graduated";
+import axios from "axios";
+
+class GraduatesList extends Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			graduates: [],
+			filterBy: 'All',
+			filterCriteria: 'All'
+		};
+	}
+
+	componentDidMount() {
+		const API_URL = process.env.REACT_APP_API_URL;
+	  axios
+			.get(`${API_URL}/empresas-egresades`)
+			.then(response => {
+				const graduates = response.data;
+				this.setState({ graduates });
+			})
+			.catch(function(error) {
+				console.log(error);
+		});
+	}
+
+	listGraduates() {
+		if (this.state.filterCriteria === 'All')
+			return this.mapGraduatedList(this.state.graduates);
+		else
+			return this.mapGraduatedList(this.factoryFilter())
+	}
+	
+	mapGraduatedList(graduatedList) {
+		return(
+			graduatedList.map((graduated, index) => {
+				return <Graduated item={graduated} key={index} />
+		}));
+	}
+
+	factoryFilter() {
+		switch (this.state.filterBy) {
+			case 'ModuleCompleted':
+				return this.filterByModuleCompleted(this.state.filterCriteria.toLowerCase());				
```

Comment:
I think it would have to be "filterByModulesCompleted", because it is possible that the graduate has more than one module completed.

Reply:
In the case Carlitos talked about i agree but instead i suggest filterByCompletedModules not ModulesCompleted because of adjective preposition used in english.

Reply:
We thought the same but, when you review the "switch case" it says "this.state.**filterBy**" and the cases explain the options, It doesnt sound good when you read "filterbyfilterbyModulesCompleted" if you change the name as you said, so i dont think that is necessary to change the name for now. :D